### DISCUSSION TITLE
FR: better support for vertical screens (including smartphone)

### DISCUSSION THREAD

**Discussion by thiswillbeyourgithub** [open-webui/open-webui#5610] [Ideas]
**Is your feature request related to a problem? Please describe.**
I use vertical screens. Besides the obvious advantages for coding, and web browsing, they are particularity great when reading LLM messages. But some features of open-webui are not working well on those:
1. The call mode is pretty bad. It has the same drawbacks as on mobile: you can't see the message and can only hear it.
![image](https://github.com/user-attachments/assets/41983bc7-9674-480f-8c53-a81b5559880c)

Which is really too bad because call mode is the only way to get a decent latency for TTS output as otherwise TTS is called only at the end of the generation #3345) (which are notoriously lengthy when using anthropic models).

Fun fact: I actually discovered that the call mode could be used on the right side of the screen only months after using it daily: 
![image](https://github.com/user-attachments/assets/7332b7e9-43c0-43ef-b000-70520c084e8c)

2. Instead of the sidebar taking half the screen even though the right half is unusable (nothing to see/do/read there) then the chat titles are cropped when they could be displayed in full. Basically use the space inside the right empty rectangle:
![image](https://github.com/user-attachments/assets/aca431df-1c81-45dc-aa3d-2f64b1560316)

**Describe the solution you'd like**
For 1. Instead of the call mode taking all the screen, make it replace the bottom elements as they are useless during call mode anyway. Basically what's in the bottom empty rectangle:
![image](https://github.com/user-attachments/assets/d04f60ee-388c-4222-8e57-e4e89d96e883)
I'm guessing that for some granny's smartphone with extra large UI elements it can be unsatisfying so if the above mentionned rectangle takes up more than 1/2 of the screen you could use the current UI instead? In any case this is bound to happen less frequently than just any vertical screen including phone as is currently the case.

For 2. As a random example I picked langchain's website which uses more horizontal space for the sidebar:

![image](https://github.com/user-attachments/assets/b422a3de-6559-485c-bbd3-afb4db19cc66)

**Describe alternatives you've considered**
[asking for admin right to inject js](https://github.com/open-webui/open-webui/issues/5420)

**Comment by thiswillbeyourgithub** [comment]
3. As requested in #4427  the code blocks are quickly unreadable on vertical screen because there's no way to toggle line wrapping.

**Comment by lvickyj** [comment]
The left and right margin are far too large on vertical screens 
![Screenshot_2024-11-09-10-37-08-316_com google android googlequicksearchbox~2](https://github.com/user-attachments/assets/721a0da2-cd01-49a7-a2f3-a3ce3545a751)
for example this is the gemini ui above which has far wider text
![Screenshot_2024-11-09-10-45-55-884_com android chrome~2](https://github.com/user-attachments/assets/64c5feb5-1229-4ebc-accb-d47390fe90bd)
The margin is significantly larger in open webui, so there should probably be a narrow screen mode

**Reply by thiswillbeyourgithub** [reply]
I totally agree
